tasks/run-file-updates/run-file-updates.yaml

```diff
@@ -25,43 +25,55 @@ spec:
     - name: request
       type: string
       description: Name of the request
-      default: "file-updates"
+      default: "process-file-updates"
     - name: synchronously
       type: string
       description: Whether to run synchronously or not
       default: "true"
     - name: pipelineRunUid
       type: string
       description: The uid of the current pipelineRun. Used as a label value when creating internal requests
+    - name: resultsDirPath
+      description: Path to the results directory in the data workspace
+      type: string
+  results:
+    - name: mergeRequestUrl
+      description: URL of MR that was created
   workspaces:
     - name: data
       description: Workspace where the file updates to apply are defined
   steps:
     - name: run-script
-      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      image: quay.io/konflux-ci/release-service-utils:7d0135b80a47cdaa225010ea1e2dff78d057c922
       script: |
-        #!/bin/sh
+        #!/bin/bash
         #
         #
-        set -e
+        set -ex
 
         # Obtain application from snapshot
         application=$(jq -rc .application $(workspaces.data.path)/$(params.snapshotPath))
         
         # Extract the key from the JSON file
         fileUpdates=$(jq -rc "$(params.jsonKey)" $(workspaces.data.path)/$(params.fileUpdatesPath))
 
-        pipelinerun_label="internal-services.appstudio.openshift.io/pipelinerun-uid"
-  
+        TASK_LABEL="internal-services.appstudio.openshift.io/group-id"
+        TASK_ID=$(context.taskRun.uid)
+        PIPELINERUN_LABEL="internal-services.appstudio.openshift.io/pipelinerun-uid"
+
+        RESULTS_FILE="$(workspaces.data.path)/$(params.resultsDirPath)/file-updates-results.json"
+        RESULTS_JSON='{"merge_requests":[]}'
+
         # Iterate over the extracted array and call the script
         fileUpdatesLength=$(jq '. | length' <<< "${fileUpdates}")
         for((i=0; i<fileUpdatesLength; i++)); do
           item=$(jq -cr ".[$i]" <<< "${fileUpdates}")
 
-          repo=$(jq -cr '.repo' <<< "${item}")
-          upstream_repo=$(jq -cr '.upstream_repo' <<< "${item}")
-          ref=$(jq -cr '.ref // "main"' <<< "${item}")
+          repo=$(jq -r '.repo' <<< "${item}")
+          upstream_repo=$(jq -r '.upstream_repo' <<< "${item}")
+          ref=$(jq -r '.ref // "main"' <<< "${item}")
           paths=$(jq -cr '.paths // "[]"' <<< "${item}")
+          file_updates_secret=$(jq -r '.file_updates_secret // "file-updates-secret"' <<< "${item}")
```

Comment:
The existing usage of this does not need the secret, right? So shouldn't be default to `""` instead?

Reply:
The existing usage does need the secret. See https://github.com/hacbs-release/app-interface-deployments/blob/main/internal-services/catalog/file-updates-process-file-updates-task.yaml#L34

We recently changed the task to allow the secret to be configurable depending on the repo...See https://github.com/hacbs-release/app-interface-deployments/pull/225

Reply:
Ah, ok, so it was hardcoded, ok, makes sense.